tools/update-tester/src/parser.rs

```diff
@@ -0,0 +1,472 @@
+        }
+    }
+    TestFile {
+        name: file_stem.to_string(),
+        stateless,
+        tests,
+    }
+}
+
+struct CodeBlockInfo {
+    kind: BlockKind,
+    transactional: bool,
+    ignore_output: bool,
+    precision_limits: HashMap<usize, usize>,
+    min_toolkit_version: Option<Version>,
+    creation: bool,
+    validation: bool,
+}
+
+#[derive(Clone, Copy)]
+enum BlockKind {
+    Sql,
+    Output,
+    Other,
+}
+
+fn parse_code_block_info(info: &str) -> CodeBlockInfo {
+    let tokens = info.split(',');
+
+    let mut info = CodeBlockInfo {
+        kind: BlockKind::Other,
+        transactional: true,
+        ignore_output: false,
+        precision_limits: HashMap::new(),
+        min_toolkit_version: None,
+        creation: false,
+        validation: false,
+    };
+
+    for token in tokens {
+        match token.trim() {
+            "ignore" => {
+                if let BlockKind::Sql = info.kind {
+                    info.kind = BlockKind::Other;
+                }
+            }
+            "non-transactional" => info.transactional = false,
+            "ignore-output" => info.ignore_output = true,
+            m if m.starts_with("min-toolkit-version") => {
+                // TODO Can we assume that version is greater than 1.10.1 since current tests don't have a min version? This means we ccan skip edge cases of 1.4/1.10.0-dev/etc.
+                info.min_toolkit_version =
+                    Some(Version::parse(token.trim_start_matches("min-toolkit-version=")).unwrap())
+            } // not great, shouldn't assume they typed in a valid version. fix later
+            "creation" => info.creation = true,
+            "validation" => info.validation = true,
+            "output" => info.kind = BlockKind::Output,
+            s if s.to_ascii_lowercase() == "sql" => info.kind = BlockKind::Sql,
+            p if p.starts_with("precision") => {
+                // syntax `precision(col: bytes)`
+                let precision_err =
+                    || -> ! { panic!("invalid syntax for `precision(col: bytes)` found `{}`", p) };
+                let arg = &p["precision".len()..];
+                if arg.as_bytes().first() != Some(&b'(') || arg.as_bytes().last() != Some(&b')') {
+                    precision_err()
+                }
+                let arg = &arg[1..arg.len() - 1];
+                let args: Vec<_> = arg.split(':').collect();
+                if args.len() != 2 {
+                    precision_err()
+                }
+                let column = args[0].trim().parse().unwrap_or_else(|_| precision_err());
+                let length = args[1].trim().parse().unwrap_or_else(|_| precision_err());
+                let old = info.precision_limits.insert(column, length);
+                if old.is_some() {
+                    panic!("duplicate precision for column {}", column)
+                }
+            }
+            _ => {}
+        }
+    }
+
+    info
+}
+
+#[cfg(test)]
+mod test {
```

Comment:
I've updated it to use `tests`. I thought that we were keeping them in the same file because they are unit tests? https://doc.rust-lang.org/book/ch11-03-test-organization.html#unit-tests
It seems like counter-agg is the only crate in toolkit that uses a `tests.rs` file right now.

Reply:
Updated release automation plan to include this in future work.